hoodie-utilities/src/main/java/com/uber/hoodie/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -235,6 +237,30 @@ public KafkaOffsetGen(TypedProperties props) {
     return offsetRanges;
   }
 
+  /***
+   * check up checkpoint offsets is valid or not, if true,  return checkpoint offsets, else return earliest offsets
+   * @param cluster,
+   * @param checkpointOffsets
+   * @param topicPartitions
+   * @return fromOffset
+   */
+  private HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> checkupValidOffsets(KafkaCluster cluster,
+                                                                                    HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> checkpointOffsets,
+                                                                                    Set<TopicAndPartition> topicPartitions ) {
+    java.util.Set<TopicAndPartition> partitions = checkpointOffsets.keySet();
```

Comment:
import java.util.Set?

Reply:
checkpointOffsets.keySet() return type of java.util.Set

Reply:
sorry. I was asking if we can simply . import it at the top, to avoid fully qualified reference here.

Reply:
in this class, it has imported ```scala.collection.immutable.Set``` and cannot ```import java.util.Set``` at the top of file.

Reply:
ah ok.. my bad